docgen/src/guides/custom-backend.md

```diff
@@ -0,0 +1,170 @@
+  }
+};
+```
+
+We use the [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) to query the server; make sure to take note of the [browser compatibility](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API#Browser_compatibility) before using it in production.
+
+> If you want to transform the data to be passed to your server, you can learn more about all the parameters that the `search()` method supports in the Search Client Specification.
+
+### Searching for facet values
+
+If your server exposes a *Search For Facet Values* endpoint, you can implement this feature in a similar way as the `search()` method with the `searchForFacetValues()` method.
+
+```javascript
+searchForFacetValues?: (requests: SearchForFacetValuesRequest[]) => Promise<{ facetHits: SearchForFacetValuesResponse[] }>
+```
+
+The implementation is the same as the `search()` method, except that we target `POST /sffv`.
+
+```javascript
+const customSearchClient = {
+  search(requests) {
+    /* ... */
+  },
+  searchForFacetValues(requests) {
+    return fetch('http://localhost:3000/sffv', {
+      method: 'post',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(requests),
+    }).then(res => res.json());
+  }
+};
+```
+
+You’ll be able to search within your refinement lists from your custom backend.
+
+> Note that you will need to set the option [`searchForFacetValues`](https://community.algolia.com/instantsearch.js/v2/widgets/refinementList.html#struct-RefinementListWidgetOptions-searchForFacetValues) to `true` in your refinement list to make it searchable.
+
+## Using the search client with InstantSearch
+
+Now, we need to tell InstantSearch to use the search client that we’ve created. This is possible with the `searchClient` option. This parameter will disable all Algolia requests coming from the frontend and will proxy them to your own backend implementation.
+
+```javascript
+const search = instantsearch({
+  indexName: 'YOUR_INDEX_NAME',
+  searchClient: customSearchClient
+});
+
+search.start();
+```
+
+That’s it! InstantSearch is now requesting your own backend and will display the UI accordingly.
+
+## Going further: enriching data from the backend
+
+Now that you’ve got InstantSearch querying your own backend before fetching results from Algolia, you could merge Algolia’s data to yours to offer your users more exhaustive results.
+
+Let’s say you want to add a download URL to each hit on your server to pass it to your frontend. You need to make a few changes to your backend:
+
+```javascript
+app.post('/search', async (req, res) => {
+  const requests = req.body;
+  const algoliaResults = await algoliaClient.search(requests);
+  const results = {
+    ...algoliaResults,
+    results: algoliaResults.results.map(result => ({
+      ...result,
+      hits: result.hits.map(hit => ({
+        ...hit,
+        downloadUrl: `https://my-website.com/files/${hit.objectID}.zip`,
```

Comment:
This example could easily be done frontend too, but there's not really a unified way of requesting data from a db or whatever. If you would have that you could do something like: 

```js
result.hits.map(async hit => ({
  ...hit,
  user: await getUser(hit.userID),
});
```

but that requires other things to be set up correctly.

Reply:
Indeed, it would make more sense to fetch data even if the given example is not exhaustive.

We can assume that the `getUser()` functions already exists or create a dummy implementation with something like [Chance.js](https://github.com/chancejs/chancejs). It's more code to digest though.

Reply:
Without assuming anything, you could have a comment that describes what getUser is, just on top of the line.